### PR TITLE
add initial enmasse chart

### PR DESCRIPTION
**Pull request by rpahli**
Signed-off-by: Rico Pahlisch <pahli88@googlemail.com>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://developercertificate.org) signed
- [ ] Chart Version bumped (if the pr is an update to an existing chart)
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[fluentd-elasticsearch]`)

### CODE REVIEW

File: charts/enmasse/templates/ClusterRole-api-server.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (axdotl) [inline charts/enmasse/templates/ClusterRole-api-server.yaml]: maybe you could add a `rbac.enabled`

File: charts/enmasse/templates/Deployment-address-space-controller.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  name: {{ include "enmasse.fullname" . }}-address-space-controller
+spec:
+  replicas: {{ .Values.addressSpaceController.replicaCount }}
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "enmasse.fullname" . }}-address-space-controller
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
```
Comment (axdotl) [inline charts/enmasse/templates/Deployment-address-space-controller.yaml]: Would it worth to support adding of annotations here (e.g. for prom scrap config)?
(same for other deployments in this PR)

File: charts/enmasse/templates/Deployment-address-space-controller.yaml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+              name: address-space-controller-config
+              optional: true
+        - name: RECHECK_INTERVAL
+          valueFrom:
+            configMapKeyRef:
+              key: recheckInterval
+              name: address-space-controller-config
+              optional: true
+        - name: IMAGE_PULL_POLICY
+          value: IfNotPresent
```
Comment (axdotl) [inline charts/enmasse/templates/Deployment-address-space-controller.yaml]: Might be configurable

File: charts/enmasse/templates/Deployment-address-space-controller.yaml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        - name: ROUTER_IMAGE
+          value: quay.io/interconnectedcloud/qdrouterd:1.9.0
```
Comment (axdotl) [inline charts/enmasse/templates/Deployment-address-space-controller.yaml]: What is the intention of the hard coded images?

File: charts/enmasse/templates/Deployment-address-space-controller.yaml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        - name: BROKER_PLUGIN_IMAGE
+          value: quay.io/enmasse/broker-plugin:0.30.1
+        - name: TOPIC_FORWARDER_IMAGE
+          value: quay.io/enmasse/topic-forwarder:0.30.1
+        - name: MQTT_GATEWAY_IMAGE
+          value: quay.io/enmasse/mqtt-gateway:0.30.1
+        - name: MQTT_LWT_IMAGE
+          value: quay.io/enmasse/mqtt-lwt:0.30.1
+        image: quay.io/enmasse/address-space-controller:0.30.1
+        imagePullPolicy: IfNotPresent
```
Comment (axdotl) [inline charts/enmasse/templates/Deployment-address-space-controller.yaml]: might be configurable

File: charts/enmasse/templates/Deployment-address-space-controller.yaml [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        name: address-space-controller
+        ports:
+        - containerPort: 8080
+          name: http
+        readinessProbe:
+          httpGet:
+            path: /healthz
+            port: http
+            scheme: HTTP
+        resources:
```
Comment (axdotl) [inline charts/enmasse/templates/Deployment-address-space-controller.yaml]: resources should be configurable

File: charts/enmasse/templates/Deployment-api-server.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+          name: http
+        - containerPort: 8443
+          name: https
+        readinessProbe:
+          httpGet:
+            path: /healthz
+            port: http
+            scheme: HTTP
+        resources:
+          limits:
```
Comment (axdotl) [inline charts/enmasse/templates/Deployment-api-server.yaml]: resources should be configurable

File: charts/enmasse/values.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+# Declare variables to be passed into your templates.
+
+operator:
+  replicaCount: 1
+apiServer:
+  replicaCount: 1
+addressSpaceController:
+  replicaCount: 1
+
+image:
```
Comment (axdotl) [inline charts/enmasse/values.yaml]: Looks like an orphaned entry from `helm create`

File: charts/enmasse/templates/Service-api-server.yaml [hunk 9]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "enmasse.fullname" . }}-api-server
+  labels:
+{{ include "enmasse.labels" . | indent 4 }}
+  annotations:
```
Comment (axdotl) [inline charts/enmasse/templates/Service-api-server.yaml]: I prefer to have extraAnnotations support on services (e.g. when using traefik)

**Comment by axdotl** [comment]
You used your private mail account for sign-off

File: charts/enmasse/templates/ClusterRole-api-server.yaml [hunk 10]
```diff
@@ -0,0 +1,19 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ClusterRole-api-server.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/ClusterRole-enmasse-operator.yaml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ClusterRole-enmasse-operator.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/ClusterRole-standard-authservice.yaml [hunk 12]
```diff
@@ -0,0 +1,12 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ClusterRole-standard-authservice.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/ClusterRoleBinding-api-server.yaml [hunk 13]
```diff
@@ -0,0 +1,16 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ClusterRoleBinding-api-server.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/ClusterRoleBinding-enmasse-operator.yaml [hunk 14]
```diff
@@ -0,0 +1,16 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ClusterRoleBinding-enmasse-operator.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/ClusterRoleBinding-standard-authservice.yaml [hunk 15]
```diff
@@ -0,0 +1,16 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ClusterRoleBinding-standard-authservice.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/Deployment-address-space-controller.yaml [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+{{ toYaml .Values.addressSpaceController.resources | indent 10 }}
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/Deployment-address-space-controller.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/Deployment-api-server.yaml [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+            path: /healthz
+            port: http
+            scheme: HTTP
+        resources:
+{{ toYaml .Values.apiServer.resources | indent 10 }}
+        volumeMounts:
+        - mountPath: /api-server-cert
+          name: api-server-cert
+          readOnly: true
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/Deployment-api-server.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/Deployment-enmasse-operator.yaml [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+        nodeAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - weight: 1
+              preference:
+                matchExpressions:
+                  - key: node-role.enmasse.io/operator-infra
+                    operator: In
+                    values:
+                      - "true"
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/Deployment-enmasse-operator.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/Role-address-space-admin.yaml [hunk 19]
```diff
@@ -0,0 +1,27 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/Role-address-space-admin.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/Role-address-space-controller.yaml [hunk 20]
```diff
@@ -0,0 +1,33 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/Role-address-space-controller.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/Role-api-server.yaml [hunk 21]
```diff
@@ -0,0 +1,18 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/Role-api-server.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/Role-enmasse-operator.yaml [hunk 22]
```diff
@@ -0,0 +1,27 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/Role-enmasse-operator.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/RoleBinding-address-space-admin.yaml [hunk 23]
```diff
@@ -0,0 +1,16 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/RoleBinding-address-space-admin.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/RoleBinding-address-space-controller.yaml [hunk 24]
```diff
@@ -0,0 +1,16 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/RoleBinding-address-space-controller.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/RoleBinding-api-server.yaml [hunk 25]
```diff
@@ -0,0 +1,16 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/RoleBinding-api-server.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/RoleBinding-enmasse-operator.yaml [hunk 26]
```diff
@@ -0,0 +1,16 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/RoleBinding-enmasse-operator.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/ServiceAccount-address-space-admin.yaml [hunk 27]
```diff
@@ -0,0 +1,8 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ServiceAccount-address-space-admin.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/ServiceAccount-address-space-controller.yaml [hunk 28]
```diff
@@ -0,0 +1,8 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ServiceAccount-address-space-controller.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/ServiceAccount-api-server.yaml [hunk 29]
```diff
@@ -0,0 +1,8 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ServiceAccount-api-server.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/ServiceAccount-enmasse-operator.yaml [hunk 30]
```diff
@@ -0,0 +1,8 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ServiceAccount-enmasse-operator.yaml]: ".Values." prefix is missing...

File: charts/enmasse/templates/ServiceAccount-standard-authservice.yaml [hunk 31]
```diff
@@ -0,0 +1,8 @@
+{{- if rbac.enabled }}
```
Comment (monotek) [inline charts/enmasse/templates/ServiceAccount-standard-authservice.yaml]: ".Values." prefix is missing...

**Pull request merged**